### ISSUE TITLE
Different loss functions

### ISSUE THREAD

**Issue by lo2aayy** [taotaonice/FaceShifter#29]
Why are the L_rec and L_att loss functions implemented different than that of the paper?

in the paper L_att, and L_rec are the L2 distance but you do squaring followed by meaning
Shouldn't they be for L2 distance: torch.norm((X - Y),2) instead of torch.mean(torch.pow(X - Y, 2))?
The difference in the loss between them is large.

**Comment by tangxuvis** [comment]
i have implemented some perfomance, can we chat on email about more details? @lo2aayy 
my email address: tangxuvis@163.com

**Comment by wvinzh** [comment]
> Why are the L_rec and L_att loss functions implemented different than that of the paper?
> 
> in the paper L_att, and L_rec are the L2 distance but you do squaring followed by meaning
> Shouldn't they be for L2 distance: torch.norm((X - Y),2) instead of torch.mean(torch.pow(X - Y, 2))?
> The difference in the loss between them is large.

I think torch.mean should be torch.sum,torch.norm((X - Y),2) is equal to torch.sqrt(torch.sum(torch.pow(X - Y, 2)))
